test/unit/data/events/scheme/get-submitted-events.test.js

```diff
@@ -0,0 +1,97 @@
+
+describe('get events', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    extractedEvent = JSON.parse(JSON.stringify(require('../../../../mocks/events/extracted')))
+    enrichedEvent = JSON.parse(JSON.stringify(require('../../../../mocks/events/enriched')))
+
+    stringifyEventData(extractedEvent)
+    stringifyEventData(enrichedEvent)
+
+    events = [extractedEvent, enrichedEvent]
+
+    mockGetClient.mockReturnValue(mockTableClient)
+    mockListEntities.mockReturnValue(events)
+  })
+
+  test('should get payment client', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockGetClient).toHaveBeenCalledTimes(1)
+  })
+
+  test('should get payment client once', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockGetClient).toHaveBeenCalledTimes(1)
+  })
+
+  test('should get payment client with payment event type', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockGetClient).toHaveBeenCalledWith(PAYMENT_EVENT)
+  })
+
+  test('should get payment events once', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockListEntities).toHaveBeenCalledTimes(1)
+  })
+
+  test('should get payment events with correlation id category', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockListEntities).toHaveBeenCalledWith({ queryOptions: { filter: mockOdata`category eq 'correlationId' and type eq 'uk.gov.defra.ffc.pay.payment.submitted'` } })
+  })
+
+  test('should return all payment events', async () => {
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(result.length).toBe(2)
+  })
+
+  test('should convert event data to json', async () => {
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(result[0].data).toEqual(extractedEvent.data)
+    expect(result[1].data).toEqual(enrichedEvent.data)
+  })
+
+  test('should return an empty array if no events', async () => {
+    mockListEntities.mockReturnValue([])
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(result.length).toBe(0)
+  })
+
+  // can this be tested because we are mocking the response from table storage?
+  test('should only return events that are of type submitted', async () => {
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    result.forEach(event => {
+      expect(event.type).toBe(PAYMENT_SUBMITTED)
+    })
+  })
+
+  // Instead can we only test what search parms the query is called with?
+  // test that it is called with the correct type
+  test('should call mockListEntities with type of submitted ', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    // what is the mock0data doing and where does it come from
+    expect(mockTableClient.listEntities).toHaveBeenCalledWith({ queryOptions: { filter: mockOdata`THIS CAN BE ANYTHING AND IT WONT FAIL${PAYMENT_SUBMITTED}` } })
```

Comment:
For more detail on odata you can read through this -> https://www.odata.org/ but as a summary it's a protocol for how to query data.

Table storage uses it and they SDK contains a handy helper function to parse js into it.

Hadn't appreciated the awkwardness of testing with it for this scenario.  I'd say make this test just confirm that the query is called correctly and create a local integration test to cover the actual usage.

I'm not sure why there aren't any local integration tests already.  Suspect an oversight.